io.github.ryanabx.flatrun.yml

```diff
@@ -0,0 +1,70 @@
+id: io.github.ryanabx.flatrun
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+base: org.flatpak.Builder.BaseApp
+base-version: "23.08"
+separate-locales: false
+command: flatrun
+finish-args:
+  - "--share=ipc"
+  - "--device=dri"
+  - "--socket=wayland"
+  - "--socket=fallback-x11"
+# Unfortunately, it's necessary for us to have this permission
+# all unsandboxed access is run through flatrun-agent.
+# flatrun-agent installs system runtimes and launches flatpaks
+# from a local repo (non system and non user).
+  - "--talk-name=org.freedesktop.Flatpak"
+build-options:
+  append-path: "/usr/lib/sdk/rust-stable/bin"
+  env:
+    CARGO_HOME: "/run/build/flatrun/cargo"
+modules:
+# Install flatpak
+  - name: flatpak
+    config-opts:
+      - "--disable-documentation"
+      - "--disable-seccomp"
+      - "--disable-sandboxed-triggers"
+      - "--disable-system-helper"
+      - "--with-system-install-dir=/var/lib/flatpak"
+    cleanup:
+      - "/etc/profile.d"
+      - "/lib/systemd"
+      - "/share/dbus-1/interfaces/org.freedesktop.*"
+      - "/share/dbus-1/services/org.freedesktop.*"
+      - "/share/gdm"
+    sources:
+      - type: archive
+        url: "https://github.com/flatpak/flatpak/releases/download/1.14.8/flatpak-1.14.8.tar.xz"
+        sha256: "1016b7327f7af87896f95465f7e5813750d3b7049a3740a1a4ddcb5fa8c5348e"
+        x-checker-data:
+          type: "json"
+          url: "https://api.github.com/repos/flatpak/flatpak/releases/latest"
+          version-query: ".tag_name"
+          url-query: "\"https://github.com/flatpak/flatpak/releases/download/\\($version)/flatpak-\\($version).tar.xz\""
+# Install flatrun
+  - name: flatrun
+    buildsystem: simple
+    build-commands:
+      # Build using BaseApp libs and binaries
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose
+      - cargo --offline build --release --verbose
+      # Cleanup BaseApp as we don't need it any more
+      - rm -rf /app/*
```

Comment:
The Flatpak Builder BaseApp includes all the libraries needed to build flatpak that aren't included in the Freedesktop SDK.

If it were up to me, I'd make the Flatpak Builder BaseApp into an SDK extension, since it includes a lot of libraries that don't come with the base SDK but are important to building flatpak (which is why Flatpak Builder needs them)

...but I don't know how hard those are to get approved.

Reply:
The baseapp was specifically made for internal consumption by infra, so it won't be switched to an extension.

Reply:
Is it still okay for me to make use of it? If not I can bring the functionality into my manifest, but I'd prefer to keep the manifest simple.

Reply:
It's ok for now.